src/Playwright/Transport/Channels/BrowserChannel.cs

```diff
@@ -91,8 +91,21 @@ internal override void OnMessage(string method, JsonElement? serverParams)
                 });
             }
 
-            args.Add("storageState", storageState);
-            args.Add("storageStatePath", storageStatePath);
+            if (!string.IsNullOrEmpty(storageStatePath))
+            {
+                if (!File.Exists(storageStatePath))
+                {
+                    throw new PlaywrightException($"The specified storage state file does not exist: {storageStatePath}");
+                }
+
+                storageState = File.ReadAllText(storageStatePath);
+            }
+
+            if (!string.IsNullOrEmpty(storageState))
+            {
+                args.Add("storageState", JsonSerializer.Deserialize<StorageState>(storageState, Helpers.JsonExtensions.DefaultJsonSerializerOptions));
+            }
```

Comment:
I would move this our of the channel. When we auto-generate the channel, it won't do this.

Reply:
This looks good, but we need a test!

Reply:
@pavelfeldman yeah, (a small) test was ~contributed~ fixed to surface this problem: https://github.com/microsoft/playwright-sharp/pull/1460 - I'm merging it now!